### PR TITLE
Rename in interpolated strings

### LINKED ISSUE #1486: Inline rename doesn't descend into interpolated strings.
``` C#
class C
{
    void Method()
    {
        var s = "Method"; // this gets updated
        var t = $"Method"; // this doesn't
    }
}
```

Renaming `Method` to `Method2` with descend into strings enabled will change `var s = "Method";` to `var s = "Method2";` but `var t = $"Method";` stays the same.

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes #1486

Update rename's "Rename in Strings" option to update interpolated strings.

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs [hunk 1]
```diff
@@ -462,7 +462,7 @@ public bool IsLiteral(SyntaxToken token)
 
         public bool IsStringLiteral(SyntaxToken token)
         {
-            return token.IsKind(SyntaxKind.StringLiteralToken);
+            return token.IsKind(SyntaxKind.StringLiteralToken, SyntaxKind.InterpolatedStringTextToken);
```
Comment (dpoeschl) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs]: I'm not sure whether IsStringLiteral should return true for InterpolatedStringTextTokens or whether we need another extension here to cover both. Opinions welcomed. Also, maybe some tests will fail and answer the question for me.

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs [hunk 2]
```diff
@@ -655,9 +655,17 @@ private SyntaxToken RenameWithinToken(SyntaxToken oldToken, SyntaxToken newToken
...
+                if (_isRenamingInStrings)
                 {
-                    newToken = RenameInStringLiteral(oldToken, newToken, SyntaxFactory.Literal);
+                    if (newToken.IsKind(SyntaxKind.StringLiteralToken))
+                    {
+                        newToken = RenameInStringLiteral(oldToken, newToken, SyntaxFactory.Literal);
+                    }
+                    else if (newToken.IsKind(SyntaxKind.InterpolatedStringTextToken))
+                    {
+                        newToken = RenameInStringLiteral(oldToken, newToken, (leadingTrivia, text, value, trailingTrivia) =>
```
Comment (dpoeschl) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs]: VB has SyntaxFactory.InterpolatedStringTextToken but C# does not. This is an API bug, right? @mattwar / @Pilchie

**Comment by dpoeschl** [comment]
Tagging reviewers: @Pilchie @jasonmalinowski @brettfo @rchande @balajikris @basoundr

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs (hunk 1 above)
Comment (dpoeschl) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs]: Tagging @brettfo because we discussed this last week.

Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs]: According to http://source.roslyn.io, rename is the only caller of that method :smiley:

File: src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs (hunk 2 above)
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs]: cc @gafter  and @AnthonyDGreen.  C# doesn't have a lot of specific token constructors in general, but I don't know if there is a reason for this difference.

File: src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb [hunk 3]
```diff
@@ -407,7 +407,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Public Function IsStringLiteral(token As SyntaxToken) As Boolean Implements ISyntaxFactsService.IsStringLiteral
-            Return CType(token.Kind, SyntaxKind) = SyntaxKind.StringLiteralToken
+            ' Return CType(token.Kind, SyntaxKind) = SyntaxKind.StringLiteralToken
```
Comment (Pilchie) [inline src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb]: Remove.

**Comment by Pilchie** [comment]
:+1:

**Comment by balajikris** [comment]
LGTM

**Comment by Pilchie** [comment]
THis looks okay to merge - the failures look like known issues that were later disabled.

**Pull request merged**